404.html

```diff
@@ -0,0 +1,191 @@
+        window.errorCode = '404';
+    </script>
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="google-site-verification" content="YD_ublu5NDiQWLZhQC5N-7eL-zo0TFpPS-l9vFqgdSI" />
+    <meta property="article:publisher" content="https://www.facebook.com/Adobe" />
+    <meta property="og:type" content="article" />
+    <meta property="og:site_name" content="Adobe Blog" />
+    <meta name="twitter:site" content="@Adobe" />
+    <meta name="twitter:creator" content="@Adobe" />
+    <meta name="twitter:card" content="summary_large_image" />
+    <meta name="serp-content-type" content="blog" />
+    <link rel="publisher" href="https://www.facebook.com/Adobe" />
+    <script>
+        const i18n = {
+            br: {
+                title: 'Estas são águas desconhecidas.',
+                subtitle: 'Desculpe, essa página não pode ser encontrada.',
+                home: 'Vá para a página inicial ›',
+                search: 'Pesquisar em Adobe.com ›',
+            },
+            de: {
+                title: 'Die Seite wurde leider nicht gefunden.',
+                subtitle: '',
+                home: 'Zur Homepage ›',
+                search: 'Adobe.com durchsuchen ›',
+            },
+            en: {
+                title: 'These are uncharted waters.',
+                subtitle: 'Sorry, that page can\'t be found.',
+                home: 'Go to homepage ›',
+                search: 'Search Adobe.com ›',
+            },
+            es: {
+                title: 'Estas son aguas inexploradas.',
+                subtitle: 'Lo siento, no se puede encontrar esa página.',
+                home: 'Ir a la página de inicio ›',
+                search: 'Buscar en Adobe.com ›',
+            },
+            fr: {
+                title: 'La page que vous recherchez ne semble pas exister.',
+                subtitle: 'Désolé, la page ne peut être trouvée.',
+                home: 'Allez à l\'accueil ›',
+                search: 'Chercher sur Adobe.com ›',
+            },
+            it: {
+                title: 'Stai navigando in acque inesplorate.',
+                subtitle: 'Pagina non trovata.',
+                home: 'Vai alla home page ›',
+                search: 'Cerca su Adobe.com ›',
+            },
+            jp: {
+                title: '申し訳ございません。ページが見つかりません。',
+                subtitle: 'Sorry, that page can\'t be found.',
+                home: 'Go to homepage ›',
+                search: 'Search Adobe.com ›',
+            },
+            ko: {
+                title: '해당 페이지가 없습니다',
+                subtitle: '죄송합니다. 페이지를 찾을 수 없습니다.',
+                home: '홈페이지 가기',
+                search: '사이트 내 검색',
+            },
+        };
+
+        window.addEventListener('load', () => {
+            const { pathname } = new URL(window.location);
+            // default english
+            const lang = Object.keys(i18n).includes(pathname.split('/')[1]) ? pathname.split('/')[1] : 'en';
```

Comment:
In light of the redirect conversation with Matt, is this still the way to do this?

Reply:
I'm not sure how this would work if Fastly redirects to `/404.html` ... you'd probably have to look at `document.referrer`